internal/testrunner/runners/pipeline/runner.go

```diff
@@ -58,32 +59,47 @@ func (r *runner) run() error {
 	}()
 
 	var failed bool
+	results := make([]testrunner.TestResult, 0)
 	for _, testCaseFile := range testCaseFiles {
+		tr := testrunner.TestResult{
+			Package:    r.options.TestFolder.Package,
+			DataStream: r.options.TestFolder.DataStream,
+		}
+		startTime := time.Now()
+
 		tc, err := r.loadTestCaseFile(testCaseFile)
 		if err != nil {
-			return errors.Wrap(err, "loading test case failed")
+			err := errors.Wrap(err, "loading test case failed")
+			tr.ErrorMsg = err.Error()
+			return results, err
 		}
-		fmt.Printf("Test case: %s\n", tc.name)
+		tr.Name = tc.name
+		results = append(results, tr)
 
 		result, err := simulatePipelineProcessing(r.options.ESClient, entryPipeline, tc)
 		if err != nil {
-			return errors.Wrap(err, "simulating pipeline processing failed")
+			err := errors.Wrap(err, "simulating pipeline processing failed")
```

Comment:
According to the current (and previous) implementation the test runner fail fast if only there is an error detected. While we have reports it would be great not to cut if off in the middle with a set of tests until one of them is marked as failed.

Reply:
Just to make sure I'm understanding correctly: you are suggesting that we should no longer fail fast now, given that we have the ability to collect test results and report them all at the end? If so, then yes, I agree with this idea – just want to confirm first that that's what you meant.

Reply:
Yes, that's the idea.

Reply:
Implemented in 7fb7adc.